src/libYARP_dev/src/yarp/dev/ISpeechSynthesizer.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2023 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef YARP_DEV_ISPEECHSYNTHESIZER_H
+#define YARP_DEV_ISPEECHSYNTHESIZER_H
+
+#include <yarp/dev/api.h>
+#include <yarp/sig/Sound.h>
+
+namespace yarp::dev {
+
+/**
+ * \ingroup dev_iface_other
+ *
+ * \brief A generic interface for speech transcription.
+ */
+class YARP_dev_API ISpeechSynthesizer
+{
+public:
+    virtual ~ISpeechSynthesizer();
+
+    /**
+     * Sets the language for speech transcription.
+     * \param language a string (code) representing the speech language (e.g. ita, eng...). Default value is "auto".
+     * \return true on success
+     */
+    virtual bool setLanguage(const std::string language="auto") = 0;
+
+    /**
+     * Gets the current language set for speech transcription.
+     * \param language the returned string (code) representing the speech language (e.g. ita, eng...). Default value is "auto".
+     * \return true on success
+     */
+    virtual bool getLanguage(std::string& language) = 0;
+
+    /**
+     * Sets the language for speech transcription.
+     * \param language a string (code) representing the speech language (e.g. ita, eng...). Default value is "auto".
```

Comment:
Probably it is useful to specify if the three-letter code uses `ISO 639-2/T`, `ISO 639-2/B` or `ISO 639-3`, see https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes ?

Reply:
My initial idea was to leave this not-defined and pass the plain string to the attached subdevice. This is because different systems might use different conventions, but I haven't checked this carefully.

However, if we want to provide a more clear and "constrained" API, then we could choose one of the three conventions that you are proposing (can you suggest which one you think is better to use?) and provide an internal conversion map (+consistency check) to the format implemented by the subdevice...

Reply:
Some extra investigation shows that:
- Amazon: https://docs.aws.amazon.com/polly/latest/dg/SupportedLanguage.html mentions "W3C language identification tags (ISO 639-3 for the language name and ISO 3166 for the country code)." 
- Google: https://cloud.google.com/text-to-speech/docs/reference/rest/v1/voices/list mentions "BCP-47 language tags".
At first glance, they seem identical to me, but I'm not sure about this.

Reply:
I would go for ISO 639-3, that seems the most similar and modern that uses three-letters codes such as `ita`, `eng`.